### ISSUE TITLE
bot stops after working for some duels

### ISSUE THREAD

**Issue by thanoskat** [will7200/Yugioh-bot#23]
2018-01-09 23:03 - DEBUG - bot.provider.Nox - Tapping at location (184,452)
2018-01-09 23:03 - DEBUG - bot.provider.Nox - Tapping at location (150,400)
2018-01-09 23:03 - INFO - bot.provider.Nox - Verifying battle
2018-01-09 23:03 - DEBUG - bot.provider.Nox - Duel-Buttons Version 1
2018-01-09 23:03 - DEBUG - bot.provider.Nox - No Auto-Duel button or Button Found
2018-01-09 23:03 - ERROR - bot - 'NoneType' object is not iterable
2018-01-09 23:03 - DEBUG - bot - <class 'TypeError'> duel_links_runtime.py 329
2018-01-09 23:03 - DEBUG - bot - Traceback (most recent call last):
  File "C:\Users\Thanos\Yugioh-bot\bot\duel_links_runtime.py", line 329, in in_main
    provider.auto()
  File "C:\Users\Thanos\Yugioh-bot\bot\providers\provider.py", line 56, in auto
    raise e
  File "C:\Users\Thanos\Yugioh-bot\bot\providers\provider.py", line 54, in auto
    self.scan()
  File "C:\Users\Thanos\Yugioh-bot\bot\providers\nox\nox.py", line 367, in scan
    battle, version = self.verify_battle()
TypeError: 'NoneType' object is not iterable

2018-01-09 23:03 - CRITICAL - bot - Provider does not have method correctly implemented cannot continue
2018-01-09 23:03 - INFO - apscheduler.executors.default - Job "DuelLinkRunTime.main.<locals>.in_main (trigger: date[2018-01-09 22:59:21 EET], next run at: 2018-01-09 22:59:21 EET)" executed successfully
2018-01-09 23:03 - INFO - bot - Value stop modified
2018-01-09 23:03 - DEBUG - bot - Timeout dump Scheduled
2018-01-09 23:03 - INFO - bot - Stopping Bot Execution
2018-01-09 23:03 - WARNING - bot - waiting for thread to stop
2018-01-09 23:03 - WARNING - bot - waiting for thread to stop
2018-01-09 23:03 - WARNING - bot - waiting for thread to stop

**Comment by apert4160** [comment]
Got a same issue.
```
2018-01-09 21:20 - DEBUG - bot.provider.Nox - LOOKING FOR BACK BUTTON, HIGH CORRERLATION
2018-01-09 21:20 - DEBUG - bot.provider.Nox - LOOKING FOR BACK BUTTON, HIGH CORRERLATION
2018-01-09 21:20 - DEBUG - bot.provider.Nox - Tapping at location (94,598)
2018-01-09 21:20 - DEBUG - bot.provider.Nox - Tapping at location (150,400)
2018-01-09 21:20 - INFO - bot.provider.Nox - Verifying battle
2018-01-09 21:20 - DEBUG - bot.provider.Nox - Duel-Buttons Version 1
2018-01-09 21:20 - DEBUG - bot.provider.Nox - No Auto-Duel button or Button Found
2018-01-09 21:20 - ERROR - bot - 'NoneType' object is not iterable
2018-01-09 21:20 - DEBUG - bot - <class 'TypeError'> duel_links_runtime.py 329
2018-01-09 21:20 - DEBUG - bot - Traceback (most recent call last):
  File "F:\WorkSpaces\Python\Yugioh-bot\bot\duel_links_runtime.py", line 329, in in_main
    provider.auto()
  File "F:\WorkSpaces\Python\Yugioh-bot\bot\providers\provider.py", line 56, in auto
    raise e
  File "F:\WorkSpaces\Python\Yugioh-bot\bot\providers\provider.py", line 54, in auto
    self.scan()
  File "F:\WorkSpaces\Python\Yugioh-bot\bot\providers\nox\nox.py", line 367, in scan
    battle, version = self.verify_battle()
TypeError: 'NoneType' object is not iterable

2018-01-09 21:20 - CRITICAL - bot - Provider does not have method correctly implemented cannot continue
2018-01-09 21:20 - INFO - apscheduler.executors.default - Job "DuelLinkRunTime.main.<locals>.in_main (trigger: date[2018-01-09 21:13:51 CST], next run at: 2018-01-09 21:13:51 CST)" executed successfully
2018-01-09 21:20 - INFO - bot - Value stop modified
2018-01-09 21:20 - DEBUG - bot - Timeout dump Scheduled
2018-01-09 21:20 - INFO - bot - Stopping Bot Execution
2018-01-09 21:20 - WARNING - bot - waiting for thread to stop
2018-01-09 21:20 - DEBUG - bot.debug_helpers - Called by: []
2018-01-09 21:20 - WARNING - bot - waiting for thread to stop
2018-01-09 21:20 - WARNING - bot - waiting for thread to stop
2018-01-09 21:20 - WARNING - bot - waiting for thread to stop
2018-01-09 21:20 - WARNING - bot - waiting for thread to stop
```

**Comment by will7200** [comment]
I know the issue the comparator to ensure that the auto duel button is present is failing, will fix.

**Comment by will7200** [comment]
This was resolved in #24, instead of using the cached stats it just ensures that auto-duel is in the expected position.
